src/cls/IPM/Repo/Oras/Definition.cls

```diff
@@ -0,0 +1,183 @@
+	Set tClient.Location = ..URL
+	Set tClient.Namespace = ..Namespace
+	Set tClient.Username = ..Username
+	Set tClient.Password = ..Password
+    Set tClient.Token = ..Token
+	Quit tClient
+}
+
+/// Returns a REST client to publish modules/applications to this server.
+Method GetPublishService() As %IPM.Repo.IPublishService
+{
+	Set tClient = ##class(%IPM.Repo.Oras.PublishService).%New()
+	Set tClient.Location = ..URL
+	Set tClient.Namespace = ..Namespace
+	Set tClient.Username = ..Username
+	Set tClient.Password = ..Password
+    Set tClient.Token = ..Token
+	Quit tClient
+}
+
+/// Outputs information about this server to the current device.
+/// Subclasses may override to show additional information.
+Method Display()
+{
+	Do ##super()
+
+	Write !,$c(9),"Deployment Enabled? ",$c(9),$Case(..DeploymentEnabled,1:"Yes",:"No")
+	
+	If (..Namespace '= "") {
+		Write !,$c(9),"Namespace: ",$c(9),$c(9),..Namespace
+	}
+	If (..Username '= "") {
+		Write !,$c(9),"Username: ",$c(9),$c(9),..Username
+	}
+	If (..Password '= "") {
+		Write !,$c(9),"Password: ",$c(9),$c(9),$Case(..Password,"":"<unset>",:"<set>")
+	}
+	If (..Token '= "") {
+		Write !,$c(9),"Token: ",$c(9),$c(9),$c(9),$Case(..Token,"":"<unset>",:"<set>")
+	}
+}
+
+/// Handles modifiers/data attributes provided from the package manager shell.
+ClassMethod OnConfigure(pInstance As %IPM.Repo.Definition, pInteractive As %Boolean = 1, ByRef pModifiers, ByRef pData) As %Status
+{
+	Set tSC = $$$OK
+	Try {
+		Set tUrl = $Get(pModifiers("url"), pInstance.URL)
+		If (..ORASURLExists(tUrl, .tId)) && (tId '= pInstance.%Id())  {
+			$$$ThrowStatus($$$ERROR($$$GeneralError, "Unable to add new ORAS server with non-unique URL."))
+		}
+		If (pInteractive && (tUrl = "")) {
+			Set tPrompt = ..#URLPromptString
+			Set tResponse = ##class(%Library.Prompt).GetString(tPrompt, .tUrl)
+			If (tResponse '= $$$SuccessResponse) {
+				$$$ThrowStatus($$$ERROR($$$GeneralError, "Operation cancelled."))
+			}
+		}
+		If $Data(pModifiers("username"), tUsername) {
+			Set pInstance.Username = tUsername
+		}
+		If $Data(pModifiers("password"), tPassword) {
+			Set pInstance.Password = tPassword
+		}
+		If $Data(pModifiers("token"), tToken) {
+			Set pInstance.Token = tToken
+		}
+		If $Data(pModifiers("publish"), tPublish) {
+			Set pInstance.DeploymentEnabled = tPublish
+		}
+		If tUrl'="" {
+			Set pInstance.URL = tUrl
+		}
+		If (pInstance.DeploymentEnabled = 1) {
+			Set tServer = ##class(%IPM.Repo.Oras.Definition).DeploymentServerOrasOpen(1)
+			If $Isobject(tServer),tServer.URL'=pInstance.URL {
+				Set tServer.DeploymentEnabled = 0
+				$$$ThrowOnError(tServer.%Save())
+			}
+		}
+		If $Data(pModifiers("namespace"), tNamespace) {
+			Set pInstance.Namespace = tNamespace
+		}
+	} Catch e {
+		Set tSC = e.AsStatus()
+	}
+	Quit tSC
+}
+
+Method GetSortOrder() As %Integer [ Private ]
+{
+	// TODO: Allow this to be changed dynamically.
+	Quit 2 + $Case(..%Id(),"":1,:..%Id())
```

Comment:
What is the need for the "2 +"? If there is no Id then this hasn't yet been saved so the sort order wouldn't be relevant anyway right?

Reply:
Honestly not too sure, but it is from IPM.Repo.Remote.Definition. I'll take a closer look.

Reply:
Looks like the "2" is related to %IPM.Repo.FileSystem.Definition:
```
Method GetSortOrder() As %Integer [ Private ]
{
	Quit 2
}
```